src/components/modes/teacher/BackgroundToggle.js

```diff
@@ -25,26 +26,51 @@ const useStyles = makeStyles(() => ({
 const BackgroundToggle = () => {
   const classes = useStyles();
   const { t } = useTranslation();
-  const dispatch = useDispatch();
 
-  const { backgroundImage } = useSelector(
-    ({ appInstance }) => appInstance.content.settings,
+  const { mutate: postAppSetting } = useMutation(
+    MUTATION_KEYS.POST_APP_SETTING,
   );
 
-  const toggleDisabled = !backgroundImage?.uri;
+  const { mutate: patchAppSetting } = useMutation(
+    MUTATION_KEYS.PATCH_APP_SETTING,
+  );
+
+  const [ backgroundToggleSetting, setBackgroundToggleSetting ] = useState(null);
+  
+  const { data: appSettings, isSuccess } = useAppSettings();
+
+  useEffect(() => {
+    if(isSuccess) {
+      const backgroundSetting = appSettings?.find(
+        ({ name }) => name === APP_SETTINGS.BACKGROUND,
+      );
+      if(backgroundSetting){
+        setBackgroundToggleSetting(appSettings?.find(
+          ({ name }) => name === APP_SETTINGS.BACKGROUND_TOGGLE,
+        ) || {
+          name: APP_SETTINGS.BACKGROUND_TOGGLE,
+          data: {
+            toggle: false,
+          }
+        });
+      }
+    }
+  }, [appSettings, isSuccess]);
+
+  const toggleDisabled = backgroundToggleSetting === null;
 
   const handleToggle = () => {
-    dispatch(
-      patchAppInstance({
-        data: {
-          backgroundImage: {
-            name: backgroundImage.name,
-            uri: backgroundImage.uri,
-            visible: !backgroundImage.visible,
-          },
-        },
-      }),
-    );
+    const newBackgroundToggleSetting = {
+      ...backgroundToggleSetting,
```

Comment:
Most probably if this is `null` it will crash, right? I would use `...(backgroundToggleSetting ?? {})` or set a right value in your `useEffect`

Reply:
As I changed the useEffect, it cannot be null:

`
setBackgroundToggleSetting(appSettings?.find(
        ({ name }) => name === APP_SETTINGS.BACKGROUND_TOGGLE,
    ) || {
        name: APP_SETTINGS.BACKGROUND_TOGGLE,
        data: {
            toggle: false,
        }
});
`
https://github.com/graasp/graasp-app-sticky-notes/blob/9c857cc58cd4d5252be159fa50e085425c4f7737/src/components/modes/teacher/BackgroundToggle.js#L48-L55

Reply:
I changed and made a default object for such cases: `...(backgroundToggleSetting ?? DEFAULT_BACKGROUND_TOGGLE)`